master/custom/steps.py

```diff
@@ -106,3 +107,16 @@ class Uninstall(ShellCommand):
     command = "chmod -R +w target/ &&  rm -rf target/"
     alwaysRun = True
     usePTY = False
+
+class UploadTestResults(steps.FileUpload):
+    warnOnFailure = True
+    haltOnFailure = False
+    flunkOnFailure = False
+    alwaysRun = True
+
+    def __init__(self):
+        super().__init__(
+                workersrc="test-results.xml",
+                masterdest=util.Interpolate("/data/www/buildbot/test-results/%(prop:buildername)s_%(prop:buildnumber)s.xml")
```

Comment:
Perhaps a directory per builder?  Or even `<branch>/<builder>/build<number>.xml`?  This does have the potential to become quite crowded if left flat.

Reply:
Done in 7e95c85